### PR TITLE
Introduce shopping list

### PR DESCRIPTION
**Pull request by jmastr**

### CODE REVIEW

**Comment by jmastr** [comment]
Note: the food list view and the shopping list view have both its very own SearchBar now, which is kind of confusing for the user.

File: qml/components-generic/JsonDownloadHelper.qml [hunk 1]
```diff
@@ -55,7 +55,7 @@ Timer {
     function loadVenueJson()
     {
         restart()
-        _fileRequest("http://www.berlin-vegan.de/app/data/GastroLocations.json")
+        _fileRequest("https://www.berlin-vegan.de/app/data/GastroLocations.json")
```
Comment (micuintus) [inline qml/components-generic/JsonDownloadHelper.qml]: Agreed!

But whether HTTPS *actually* means security is debatable ;)

File: qml/harbour-berlin-vegan.qml [hunk 2]
```diff
@@ -62,6 +62,31 @@ ApplicationWindow
...
+        id: gjsonShoppingModelCollection
+
+        property string searchString: ""
+        property bool   loaded: false
+
+        onSearchStringChanged: reFilter()
+
+        model: jsonShoppingModel
+
+        comparator: function lessThan(a, b) {
```
Comment (micuintus) [inline qml/harbour-berlin-vegan.qml]: Regarding your question in the commit message: Yes, QML offers inheritance. If, in your example, you add a file
```MyCollection.qml``` and put something in it like
```
Collection {
        property string searchString: ""
        property bool   loaded: false

        onSearchStringChanged: reFilter()

        model: jsonShoppingModel

        comparator: function lessThan(a, b) {
...
```
you would be inheriting from Collection. If putting this in an extra file is too cumbersome for you, you can also use this ```Component``` type to declare a component inline within a file: http://doc.qt.io/qt-5/qml-qtqml-component.html

And yes, answering your second question, that whole collection business is one of the thing that we want to move over to C++ in the mind run.

**Comment by micuintus** [comment]
Very nice, job!

Sorry I took so long and didn't merge this yet. I was (and still am) not 100% sure how to deal with the menu situation --- on Sailfish, but also in general.

The menu in Sailfish feels really weird now ;). I'd love to find a way that these two menu items (food list and shopping list) don't push a new page, but replace the current page and both have the (global) pull down menu. I already tried to implement something, but that isn't working (yet).

For the long run, I am thinking about having a concept of a toggle-able menu item, which holds two menu items, but only the non-active appears in the menu and you can toogle between them.
 
On a non Sailfish specific note I wonder whether we actually want to have completely new page for these kind of entries, at all? Or do we rather only want to switch the collection when you hit one of the menu buttons, but leave everything else (search bar, etc.) the same?

**Comment by jmastr** [comment]
> Or do we rather only want to switch the collection when you hit one of the menu buttons, but leave everything else (search bar, etc.) the same?

I would prefer that solution, i.e. create just one VenueList and update the collection in it (and also clear the SearchBar, if a user switches), because there will be a third list: Favourites.

**Pull request merged**